graviton/abi_strategy.py

```diff
@@ -4,16 +4,28 @@
 TODO: Leverage Hypothesis!
 """
 from abc import ABC, abstractmethod
+from enum import Enum, auto
 import random
 import string
-from typing import List, Optional, Sequence, cast
+from typing import List, Optional, Sequence, Type, cast
 
 from algosdk import abi, encoding
 
+
 from graviton.models import PyTypes
 
 
 class ABIStrategy(ABC):
+    """
+    TODO: when incorporating hypothesis strategies, we'll need a more holistic
+    approach that looks at relationships amongst various args.
+    Current approach only looks at each argument as a completely independent entity.
```

Comment:
no action needed, I see hypothesis cross the PR, together with TODO, are you referring to hypothesis https://hypothesis.readthedocs.io/en/latest/data.html?

Reply:
Yes. Here's a nice video intro about it: https://www.youtube.com/watch?v=mkgd9iOiICc&ab_channel=ArjanCodes